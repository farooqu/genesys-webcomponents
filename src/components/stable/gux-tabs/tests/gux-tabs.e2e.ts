import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { a11yCheck } from '../../../../../tests/e2eTestUtils';

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();
  await page.addScriptTag({
    path: 'node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-tabs', () => {
  const html = `
    <gux-tabs lang="en">
      <gux-tab-list slot="tab-list">
          <gux-tab tab-id="2-1"><span>Tab Header 1</span></gux-tab>
          <gux-tab tab-id="2-2"><span>Tab Header 2</span></gux-tab>
          <gux-tab tab-id="2-3"><span>Tab Header 3</span></gux-tab>
          <gux-tab gux-disabled tab-id="2-4"
            ><span>Tab Header 4</span></gux-tab
          >
          <gux-tab gux-disabled tab-id="2-5"
            ><span>Tab Header 5</span></gux-tab
          >
      </gux-tab-list>
      <gux-tab-panel tab-id="2-1">Tab content 1</gux-tab-panel>
      <gux-tab-panel tab-id="2-2">Tab content 2</gux-tab-panel>
      <gux-tab-panel tab-id="2-3">Tab content 3</gux-tab-panel>
      <gux-tab-panel tab-id="2-4">Tab content 4</gux-tab-panel>
      <gux-tab-panel tab-id="2-5">Tab content 5</gux-tab-panel>
    </gux-tabs>
`;
  describe('#render', () => {
    it('renders', async () => {
      const page = await newNonrandomE2EPage({ html });
      const element = await page.find('gux-tabs');

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('does not render the scroll buttons when tabs fit container', async () => {
      const page = await newNonrandomE2EPage({ html });
      const scrollButtons = await page.findAll('.gux-scroll-button');
      await a11yCheck(page);

      expect(scrollButtons.length).toBe(0);
    });

    it('renders scroll buttons when tabs overflow container', async () => {
      const restrictedWidthHtml = `<div style="width: 500px">${html}</div>`;
      const page = await newNonrandomE2EPage({ html: restrictedWidthHtml });
      const scrollButtons = await page.findAll('.gux-scroll-button');
      await a11yCheck(page);

      expect(scrollButtons.length).toBe(2);
    });
  });

  describe('#interactions', () => {
    it('should change tabpanel content when tab is changed', async () => {
      const page = await newNonrandomE2EPage({ html });
      const tabTarget = await page.find('gux-tab[tab-id="2-2"]');
      const tabPanel = await page.find('gux-tab-panel[tab-id="2-2"]');
      const spyOnActivePanelChangeEvent = await tabPanel.spyOnEvent(
        'guxactivepanelchange'
      );

      await tabTarget.click();
      await page.waitForChanges();

      expect(spyOnActivePanelChangeEvent.length).toBe(1);
      expect(spyOnActivePanelChangeEvent.events[0].detail).toBe('2-2');
    });
    it('should not change tabpanel content when tab is disabled', async () => {
      const page = await newNonrandomE2EPage({ html });
      const tabTarget = await page.find('gux-tab[tab-id="2-4"]');
      const tabPanel = await page.find('gux-tab-panel[tab-id="2-4"]');
      const spyOnActivePanelChangeEvent = await tabPanel.spyOnEvent(
        'guxactivepanelchange'
      );

      await tabTarget.click();
      await page.waitForChanges();

      expect(spyOnActivePanelChangeEvent.length).toBe(0);
    });
  });
});
