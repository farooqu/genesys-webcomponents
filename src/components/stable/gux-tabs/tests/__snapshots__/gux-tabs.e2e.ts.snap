// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tabs renders 1`] = `" <gux-tab tab-id=\\"1\\" tab-icon-name=\\"lock\\" class=\\"hydrated\\"><!----> <button class=\\"gux-tab show\\" role=\\"button\\"><div class=\\"tab-icon-container\\"><gux-icon icon-name=\\"lock\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M12.5 7H12V5c0-1.763-1.166-3.244-2.758-3.77l-.005-.001a3.826 3.826 0 00-.573-.149c-.216-.041-.436-.067-.664-.067s-.448.026-.664.067a3.826 3.826 0 00-.578.151C5.166 1.756 4 3.237 4 5v2h-.5C2.671 7 2 7.67 2 8.5v6A1.5 1.5 0 003.5 16h9a1.5 1.5 0 001.5-1.5v-6A1.5 1.5 0 0012.5 7zM6 5a2 2 0 014 0v2H6V5z\\"></path></svg></div></gux-icon></div><span class=\\"tab-title\\"><span slot=\\"title\\"> Hello World </span></span></button><button id=\\"1\\" class=\\"tab-dropdown-container\\"><gux-icon icon-name=\\"menu-kebab-v\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M8 4.5a1.5 1.5 0 10-.001-3.001A1.5 1.5 0 008 4.5zm0 2a1.5 1.5 0 10.001 3.001A1.5 1.5 0 008 6.5zm0 5a1.5 1.5 0 10.001 3.001A1.5 1.5 0 008 11.5z\\"></path></svg></div></gux-icon></button><gux-popover hidden=\\"\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"top\\" class=\\"hydrated\\" style=\\"position: absolute; left: 0px; top: auto; margin: 0px; right: auto; bottom: 0px; transform: translate(56px, -585px);\\"><!----><div class=\\"gux-popover-wrapper\\"><div class=\\"gux-arrow\\" data-popper-arrow></div><div class=\\"gux-popover-content\\"><div><span slot=\\"dropdown-options\\"> <gux-tab-dropdown-option option-id=\\"1\\" icon-name=\\"edit\\" onclick=\\"notify(event)\\" class=\\"hydrated\\"><!----><button class=\\"tab-dropdown-option\\"><gux-icon icon-name=\\"edit\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M1 12.189V15h2.812l-1.951-1.95zM14.561 2.146l-.707-.707a1.5 1.5 0 00-2.119 0L11 2.173l-9.2 9.2L4.63 14.2l8.87-8.872c.826-.826.938-.936 1.061-1.061a1.5 1.5 0 000-2.121z\\"></path></svg></div></gux-icon><span> Edit </span></button></gux-tab-dropdown-option> <gux-tab-dropdown-option option-id=\\"2\\" icon-name=\\"clone\\" onclick=\\"notify(event)\\" class=\\"hydrated\\"><!----><button class=\\"tab-dropdown-option\\"><gux-icon icon-name=\\"clone\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M2 3H1v12h12v-1H2zm1-2v12h12V1zm9.6 7.2h-2.4v2.4H7.8V8.2H5.4V5.8h2.4V3.4h2.4v2.4h2.4z\\"></path></svg></div></gux-icon><span> Clone </span></button></gux-tab-dropdown-option> <gux-tab-dropdown-option option-id=\\"3\\" icon-name=\\"share\\" onclick=\\"notify(event)\\" class=\\"hydrated\\"><!----><button class=\\"tab-dropdown-option\\"><gux-icon icon-name=\\"share\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M12.5 10.1c-.56 0-1.068.2-1.477.521L6.871 8.288c.011-.096.029-.189.029-.288s-.018-.192-.029-.288l4.153-2.333c.408.321.916.521 1.476.521a2.4 2.4 0 10-2.4-2.4c0 .101.018.197.03.295l-4.148 2.33A2.38 2.38 0 004.5 5.6a2.4 2.4 0 100 4.8c.562 0 1.072-.201 1.481-.525l4.148 2.33c-.012.098-.03.194-.03.295a2.4 2.4 0 102.401-2.4z\\"></path></svg></div></gux-icon><span> Share </span></button></gux-tab-dropdown-option> <gux-tab-dropdown-option option-id=\\"4\\" icon-name=\\"download\\" onclick=\\"notify(event)\\" class=\\"hydrated\\"><!----><button class=\\"tab-dropdown-option\\"><gux-icon icon-name=\\"download\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M2 13v1h12v-1zm6.006-1.006l.006.006 4.573-4.586L11.171 6 9.006 8.166V2h-2v6.166L4.841 6 3.427 7.414 8 12z\\"></path></svg></div></gux-icon><span> Download </span></button></gux-tab-dropdown-option> </span></div></div></div></gux-popover></gux-tab> <gux-tab tab-id=\\"2\\" tab-icon-name=\\"lock\\" class=\\"hydrated\\"><!----> <button class=\\"gux-tab show\\" role=\\"button\\"><div class=\\"tab-icon-container\\"><gux-icon icon-name=\\"lock\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M12.5 7H12V5c0-1.763-1.166-3.244-2.758-3.77l-.005-.001a3.826 3.826 0 00-.573-.149c-.216-.041-.436-.067-.664-.067s-.448.026-.664.067a3.826 3.826 0 00-.578.151C5.166 1.756 4 3.237 4 5v2h-.5C2.671 7 2 7.67 2 8.5v6A1.5 1.5 0 003.5 16h9a1.5 1.5 0 001.5-1.5v-6A1.5 1.5 0 0012.5 7zM6 5a2 2 0 014 0v2H6V5z\\"></path></svg></div></gux-icon></div><span class=\\"tab-title\\"><span slot=\\"title\\"> Hello World 2 </span></span></button></gux-tab> "`;
